success/index.js

```diff
@@ -1,66 +1,213 @@
-(() => {
-  // Parse query string from URL
-  const qs = ((a) => {
-    if (a === '') return {};
-    var b = {}
-    for (var i = 0; i < a.length; ++i) {
-      var p = a[i].split('=')
-      if (p.length !== 2) continue
-      b[p[0]] = decodeURIComponent(p[1].replace(/\+/g, ' '))
+(async () => {
+  /* Fetch and Parse Location */
+  const hpAdminUrl = (hpAdminQueryString => {
+    if (hpAdminQueryString === '') return {}
+    const queryStringKv = {}
+    for (let i = 0; i < hpAdminQueryString.length; ++i) {
+      const queryStringParam = hpAdminQueryString[i].split('=')
+      if (queryStringParam.length !== 2) continue
+      queryStringKv[queryStringParam[0]] = decodeURIComponent(queryStringParam[1].replace(/\+/g, ' '))
     }
-    return b
+    return queryStringKv
   })(window.location.search.substr(1).split('&'))
 
+  if (hpAdminUrl && !hpAdminUrl.url) {
+    console.log('No HoloPort URL found in the query')
```

Comment:
not sure which, but we either want to throw an error here or just fail silently. Putting this in a log is only useful for development.